warzone/Cards.cpp

```diff
@@ -0,0 +1,442 @@
+	 */
+	SpyCard& SpyCard::operator=(const SpyCard& card)
+	{
+		return new SpyCard(card);
+	}
+
+	/**
+	 * Play function
+	 * 
+	 */
+	void SpyCard::play()
+	{
+		cout << "Playing SpyCard" << endl;
+	}
+
+	// Bomb Cards
+	BombCard::BombCard(): Card()
+	{
+		this->name = new std::string("BombCard");
+	}
+
+	BombCard::BombCard(BombCard* bombCard) : Card(bombCard)
+	{
+	}
+
+	BombCard::~BombCard()
+	{
+	}
+
+	BombCard& BombCard::operator=(const BombCard& card)
+	{
+		return new BombCard(card);
+	}
+
+	/**
+	 * Play function
+	 *
+	 */
+	void BombCard::play()
+	{
+		cout << "Playing BombCard" << endl;
+	}
+
+	// Reinforcement Cards
+	ReinforcementCard::ReinforcementCard() : Card()
+	{
+		name = new std::string("ReinforcementCard");
+	}
+
+	ReinforcementCard::ReinforcementCard(ReinforcementCard* rCard) : Card(rCard)
+	{
+	}
+
+	ReinforcementCard::~ReinforcementCard()
+	{
+	}
+
+	ReinforcementCard& ReinforcementCard::operator=(const ReinforcementCard& card)
+	{
+		return new ReinforcementCard(card);
+	}
+
+	/**
+	 * Play function
+	 *
+	 */
+	void ReinforcementCard::play()
+	{
+		cout << "Playing ReinforcementCard" << endl;
+	}
+
+	// Blockade Cards
+	BlockadeCard::BlockadeCard(): Card()
+	{
+		name = new std::string("BlockadeCard");
+	}
+
+	
+	BlockadeCard::BlockadeCard(BlockadeCard* bCard) : Card(bCard)
+	{
+	}
+
+	BlockadeCard& BlockadeCard::operator=(const BlockadeCard& card)
+	{
+		return new BlockadeCard(card);
+	}
+
+	BlockadeCard::~BlockadeCard()
+	{
+
+	}
+
+	/**
+	 * Play function
+	 *
+	 */
+	void BlockadeCard::play()
+	{
+		cout << "Playing BlockadeCard" << endl;
+	}
+
+	// Airlift Cards
+	AirliftCard::AirliftCard() : Card()
+	{
+		name = new std::string("AirliftCard");
+	}
+
+	AirliftCard::AirliftCard(AirliftCard* aCard) : Card(aCard)
+	{
+	}
+
+	AirliftCard::~AirliftCard()
+	{
+
+	}
+
+	AirliftCard& AirliftCard::operator=(const AirliftCard& card)
+	{
+		return new AirliftCard(card);
+	}
+
+	/**
+	 * Play function
+	 *
+	 */
+	void AirliftCard::play()
+	{
+		cout << "Playing AirliftCard" << endl;
+	}
+
+	// Diplomacy Cards
+	DiplomacyCard::DiplomacyCard() : Card()
+	{
+		name = new std::string("DiplomacyCard");
+	}
+
+	DiplomacyCard::DiplomacyCard(DiplomacyCard* dCard) : Card(dCard)
+	{
+	}
+
+	DiplomacyCard::~DiplomacyCard()
+	{
+
+	}
+
+	DiplomacyCard& DiplomacyCard::operator=(const DiplomacyCard& card)
+	{
+		return new DiplomacyCard(card);
+	}
+
+	/**
+	 * Play function
+	 *
+	 */
+	void DiplomacyCard::play()
+	{
+		cout << "Playing DiplomacyCard" << endl;
+	}
+
+	// Deck functions
+	Deck::Deck()
+	{
+		this->fullDeck = new std::vector<Card*>;
+		this->drawPile = new std::vector<int*>;
+		return;
+	}
+
+	Deck::Deck(Deck* deck)
+	{
+		this->fullDeck = new vector<Card*>;
+		this->drawPile = new vector<int*>;
+
+		for (int i = 0; i < deck->fullDeck->size(); i++)
+		{
+			fullDeck->push_back(deck->fullDeck->at(i));
+		}
+		for (int i = 0; i < deck->drawPile->size(); i++)
+		{
+			drawPile->push_back(deck->drawPile->at(i));
+		}
```

Comment:
I think we need to call the copy constructor of the card in the other deck, we can't just copy the pointer else it's not a deep copy.